lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -119,6 +119,34 @@ class StableHLOToTTIRReduceOpConversionPattern
   }
 };
 
+class StableHLOToTTIRTransposeOpConversionPattern
+    : public OpConversionPattern<mlir::stablehlo::TransposeOp> {
+  using OpConversionPattern<mlir::stablehlo::TransposeOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(mlir::stablehlo::TransposeOp srcOp,
+                  mlir::stablehlo::TransposeOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto outputType = mlir::cast<RankedTensorType>(srcOp.getResult().getType());
+    auto outputTensor = rewriter.create<tensor::EmptyOp>(
+        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+    assert(adaptor.getPermutation().size() <= 2 &&
+           "TTIR only supports maximum of two dimensional transposeOp.");
```

Comment:
Size should be strictly 2 since you are later on passing `adaptor.getPermutation()[0]` and `adaptor.getPermutation()[1]` as arguments to `ttir::TransposeOp` creation. Please also update the assert message accordingly.